src/boolean_result.cpp

```diff
@@ -809,6 +810,40 @@ Manifold::Impl Boolean3::Result(OpType op) const {
   PRINT(n12 << " new verts from edgesP -> facesQ");
   PRINT(n21 << " new verts from facesP -> edgesQ");
 
+  const int numProp = std::max(inP_.NumProp(), inQ_.NumProp());
+  if (numProp > 0) {
+    // Ideally, switch over from triProperties vec to Halfedge.startProp - then
+    // the prop index will be available in the append edges functions below.
+    Vec<std::atomic<int>> prop2VertP(inP_.NumPropVert());
+    for_each_n(autoPolicy(inP_.NumTri()), countAt(0), inP_.NumTri(),
+               [&](size_t tri) {
+                 for (const int i : {0, 1, 2}) {
+                   prop2VertP[inP_.meshRelation_.triProperties[tri][i]].store(
+                       inP_.halfedge_[3 * tri + i].startVert);
+                 }
+               });
+    Vec<std::atomic<int>> prop2VertQ(inQ_.NumPropVert());
+    for_each_n(autoPolicy(inQ_.NumTri()), countAt(0), inQ_.NumTri(),
+               [&](size_t tri) {
+                 for (const int i : {0, 1, 2}) {
+                   prop2VertQ[inQ_.meshRelation_.triProperties[tri][i]].store(
+                       inQ_.halfedge_[3 * tri + i].startVert);
+                 }
+               });
+    // Scan the properties inclusion numbers, e.g. + i03[prop2VertP[prop]].
+    //
+    // New verts create either 2 or 3 new properties: one for the face, plus one
+    // for an edge where start and end verts share properties, or plus two for
+    // an edge that doesn't share both properties.
```

Comment:
This is the answer to question 2-i. It might be clearer if I said "prop verts" instead of "properties" - that's what I mean.